src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -7,12 +7,100 @@
 #include <iterator>
 #include <numeric>
 #include <random>
+#include <thread>
 #include "file_io_read_micro_benchmark.hpp"
 #include "micro_benchmark_basic_fixture.hpp"
 
 namespace hyrise {
 
-BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, READ_NON_ATOMIC_SEQUENTIAL)(benchmark::State& state) {  // open file
+void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
+  lseek(fd, from * uint32_t_size, SEEK_SET);
+  Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
+         fail_and_close_file(fd, "Read error: ", errno));
+}
+
+void read_data_randomly_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                                   const std::vector<uint32_t>& random_indices) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+
+  lseek(fd, 0, SEEK_SET);
+  // TODO Randomize inidzes to not read all the data but really randomize the reads to read same amount but incl possible duplicates
+  for (auto index = from; index < to; ++index) {
+    lseek(fd, uint32_t_size * random_indices[index], SEEK_SET);
+    Assert((read(fd, read_data_start + index, uint32_t_size) == uint32_t_size),
+           fail_and_close_file(fd, "Read error: ", errno));
+  }
+}
+
+void read_data_using_pread(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
+  lseek(fd, from * uint32_t_size, SEEK_SET);
+  Assert((pread(fd, read_data_start + from, bytes_to_read, from * uint32_t_size) == bytes_to_read),
+         fail_and_close_file(fd, "Read error: ", errno));
+}
+
+void read_data_randomly_using_pread(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                                    const std::vector<uint32_t>& random_indices) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+
+  lseek(fd, 0, SEEK_SET);
+  // TODO Randomize inidzes to not read all the data but really randomize the reads to read same amount but incl possible duplicates
+  for (auto index = from; index < to; ++index) {
+    Assert((pread(fd, read_data_start + index, uint32_t_size, uint32_t_size * random_indices[index]) == uint32_t_size),
```

Comment:
Why do we not use another lseek within the loop here, as in read_data_randomly_using_read? Do we even need lseek always? Because according to the docs:

pread() reads up to count bytes from file descriptor fd at offset
       offset (from the start of the file) into the buffer starting at
       buf.  The file offset is not changed.